iocore/dns/I_DNSProcessor.h

```diff
@@ -151,25 +167,31 @@ extern DNSProcessor dnsProcessor;
 inline Action *
 DNSProcessor::getSRVbyname(Continuation *cont, const char *name, Options const &opt)
 {
-  return getby(name, 0, T_SRV, cont, opt);
+  return getby(std::string_view(name), T_SRV, cont, opt);
+}
+
+inline Action *
+DNSProcessor::getSRVbyname(Continuation *cont, std::string_view name, Options const &opt)
+{
+  return getby(name, T_SRV, cont, opt);
 }
 
 inline Action *
 DNSProcessor::gethostbyname(Continuation *cont, const char *name, Options const &opt)
 {
-  return getby(name, 0, T_A, cont, opt);
+  return getby(std::string_view(name), T_A, cont, opt);
 }
 
 inline Action *
-DNSProcessor::gethostbyname(Continuation *cont, const char *name, int len, Options const &opt)
+DNSProcessor::gethostbyname(Continuation *cont, std::string_view name, Options const &opt)
 {
-  return getby(name, len, T_A, cont, opt);
+  return getby(name, T_A, cont, opt);
 }
 
 inline Action *
 DNSProcessor::gethostbyaddr(Continuation *cont, IpAddr const *addr, Options const &opt)
 {
-  return getby(reinterpret_cast<const char *>(addr), 0, T_PTR, cont, opt);
+  return getby(*addr, T_PTR, cont, opt);
 }
```

Comment:
`addr` is dereferenced without checking for nullptr. Should addr be taken as a const reference?

Reply:
Let me try that.

Reply:
Also, this change just moves the failure point. If you look at "DNS.cc:439" you can see the original code dereferences the pointer without checking for `nullptr`. This change simply makes it fail earlier in the call stack.

Reply:
Yeah, to be clear I was assuming that we expected logically that this would not be nullptr. I was just wondering whether it made sense, that being the case, to make this a reference.

Reply:
I changed the struct to have a pointer instead of an instance. Not sure why I had an instance in the first place - that forces an unnecessary copy.